src/engine/strat_engine/filesystem.rs

```diff
@@ -150,19 +150,38 @@ impl StratFilesystem {
     }
 
     /// Get the mount_point for this filesystem
-    /// TODO Replace this code with something less brittle
     pub fn get_mount_point(&self) -> EngineResult<Option<PathBuf>> {
-        let output = Command::new("df")
-            .arg("--output=target")
-            .arg(&self.devnode())
-            .output()?;
-        if output.status.success() {
-            let output_str = String::from_utf8_lossy(&output.stdout);
-            return Ok(output_str.lines().last().map(PathBuf::from));
+        if let Some(search) = self.devnode().to_str() {
```

Comment:
Generally you try to use the functions on Option, so, instead of the let-if, you would do:

```
let search = self.devnode().to_str().ok_or_else(|| EngineError::Engine(ErrorEnum::Error, format!("Unable to represent devnode as string...")))?;
let m_iter = MountIter::new_from_proc().map_err(|_| EngineError::Engine(ErrorEnum::Error, "Error opening /proc/mounts..."))?;
for mp in m_iter {
  // probably more to be done here....
}
```